janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java

```diff
@@ -44,111 +26,109 @@
+import java.net.Socket;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
-    protected static final String BUILD_DIR;
-    protected static final String EXPECT_DIR;
-    protected static final String ZIPFILE_PATH;
-    protected static final String ZIPFILE_EXTRACTED;
-    protected static final String ZIPFILE_FULL_PATH;
-    protected static final String ZIPFILE_FULL_EXTRACTED;
+import static org.apache.tinkerpop.gremlin.driver.ser.AbstractMessageSerializer.TOKEN_IO_REGISTRIES;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTimeout;
 
-    static {
-        Properties props;
+public abstract class AbstractJanusGraphAssemblyIT extends JanusGraphAssemblyBaseIT {
 
-        try {
-            props = new Properties();
-            java.io.FileReader fr = new FileReader(Joiner.on(File.separator).join(new String[] { "target", "test-classes", "target.properties" }));
-            props.load(fr);
-            fr.close();
-        } catch (IOException e) {
-            throw new AssertionError(e);
-        }
+    protected abstract String getConfigPath();
+
+    protected abstract String getServerConfigPath();
 
-        BUILD_DIR    = props.getProperty("build.dir");
-        EXPECT_DIR   = props.getProperty("expect.dir");
-        ZIPFILE_PATH = props.getProperty("zipfile.path");
-        ZIPFILE_EXTRACTED = ZIPFILE_PATH.substring(0, ZIPFILE_PATH.length() - 4);
-        ZIPFILE_FULL_PATH = props.getProperty("zipfile-full.path");
-        ZIPFILE_FULL_EXTRACTED = ZIPFILE_FULL_PATH.substring(0, ZIPFILE_FULL_PATH.length() - 4);
+    protected abstract String getGraphName();
 
-        Properties p = new Properties();
-        p.put("file.resource.loader.path", EXPECT_DIR);
-        Velocity.init(p);
+    protected Boolean supportsIndex() {
+        return false;
     }
 
-    protected void testSimpleGremlinSession(String graphConfig, String graphToString, boolean full) throws Exception {
-        unzipAndRunExpect("single-vertex.expect.vm", graphConfig, graphToString, full, false);
+    @Test
+    public void testSimpleGremlinSession() throws Exception {
+        unzipAndRunExpect("single-vertex.expect.vm", getConfigPath(), getGraphName(), false, false);
     }
 
-    private static boolean serverListening(String host, int port)
-    {
-        Socket s = null;
-        try
-        {
-            s = new Socket(host, port);
-            return true;
-        }
-        catch (Exception e)
-        {
-            return false;
-        }
-        finally
-        {
-            if(s != null)
-                try {s.close();}
-                catch(Exception e){}
+    @Test
+    @Tag(PkgTestCategory.FULL_TESTS)
+    public void testSimpleGremlinSessionFull() throws Exception {
+        unzipAndRunExpect("single-vertex.expect.vm", getConfigPath(), getGraphName(), true, false);
```

Comment:
@li-boxuan This test wasn't activated before.

Reply:
I can see "testSimpleGremlinSessionFull" in the old tests. Am I missing something?

Reply:
I will rework the naming
I'm pretty sure the file single-vertex.expect was executed.

Reply:
> I'm pretty sure the file single-vertex.expect was executed.

You mean "was executed" or "was not executed"? In the old tests, I see a few invocations of method `testSimpleGremlinSession` which calls `unzipAndRunExpect("single-vertex.expect.vm", graphConfig, graphToString, full, false);` My confusion is what exactly was missing before.